File: modules/core/src/main/java/de/deutschebahn/bahnhoflive/ui/search/StationSearchViewHolder.java
package de.deutschebahn.bahnhoflive.ui.search;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CompoundButton;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.LayoutRes;

import de.deutschebahn.bahnhoflive.R;
import de.deutschebahn.bahnhoflive.ui.ViewHolder;
import de.deutschebahn.bahnhoflive.view.CompoundButtonChecker;

public class StationSearchViewHolder extends ViewHolder<SearchResult>
        implements CompoundButton.OnCheckedChangeListener {


    public static final String TAG = StationSearchViewHolder.class.getSimpleName();

    private final TextView nameView;
    private final CompoundButtonChecker favoriteView;
    private final ImageView iconView;

    public StationSearchViewHolder(ViewGroup container, @LayoutRes int layout) {
        this(LayoutInflater.from(container.getContext()).inflate(layout, container, false));
    }

    public StationSearchViewHolder(View view) {
        super(view);

        nameView = itemView.findViewById(R.id.name);

        iconView = itemView.findViewById(R.id.icon);

        favoriteView = new CompoundButtonChecker(itemView.findViewById(R.id.favorite_indicator), this);
    }

    @Override
    protected void onBind(SearchResult item) {
        nameView.setText(item.getTitle());
        iconView.setImageResource(item.getIcon());
        iconView.setContentDescription(iconView.getResources().getText(
                item.isLocal() ? R.string.sr_stop_type_local : R.string.sr_stop_type_db));
        favoriteView.setChecked(item.isFavorite());
    }

    @Override
    public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
        final SearchResult searchResult = getItem();
        if (searchResult != null) { // item probably isn't null, but prevent a crash anyways
            searchResult.setFavorite(isChecked);
        }
    }
}

File: modules/core/src/main/java/de/deutschebahn/bahnhoflive/ui/search/StationSearchViewHolder.kt
/*
 * SPDX-FileCopyrightText: 2020 DB Station&Service AG <bahnhoflive-opensource@deutschebahn.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
package de.deutschebahn.bahnhoflive.ui.search

import android.view.View
import android.widget.CompoundButton
import android.widget.ImageView
import android.widget.TextView
import de.deutschebahn.bahnhoflive.R
import de.deutschebahn.bahnhoflive.ui.ViewHolder
import de.deutschebahn.bahnhoflive.view.CompoundButtonChecker

class StationSearchViewHolder(view: View) : ViewHolder<SearchResult?>(view), CompoundButton.OnCheckedChangeListener {

    private val nameView: TextView = itemView.findViewById(R.id.name)
    private val favoriteView: CompoundButtonChecker =
        CompoundButtonChecker(itemView.findViewById(R.id.favorite_indicator), this)
    private val iconView: ImageView = itemView.findViewById(R.id.icon)

    override fun onBind(item: SearchResult?) {
        nameView.text = item!!.title
        iconView.setImageResource(item.icon)
        nameView.contentDescription = item.title.toString() + ", " + nameView.resources.getText(
            if (item.isLocal) R.string.sr_stop_type_local else R.string.sr_stop_type_db
        )
        favoriteView.isChecked = item.isFavorite
    }

    override fun onCheckedChanged(buttonView: CompoundButton, isChecked: Boolean) {
        val searchResult = item
        if (searchResult != null) { // item probably isn't null, but prevent a crash anyways
            searchResult.isFavorite = isChecked
        }
    }

    companion object {
        val TAG: String = StationSearchViewHolder::class.java.simpleName
    }
}
